documentation/development-docs/design-guidelines/piping-best-practices.md

```diff
@@ -1,143 +1,8 @@
-# Azure PowerShell Piping Scenarios
-
-There are two main scenarios that we wish to enable in cmdlets for Azure PowerShell:
```

Comment:
I'd rather move the `Azure PowerShell Piping Scenarios` section to a separate doc and link it from the `Piping in Azure PowerShell` section.  Reasons: To discourage users from using `ValueFromPipelineByPropertyName` except for `-ResourceId`

Reply:
Though the resource id parameter set is the main scenario that will use `ValueFromPipelineByPropertyName`, there are cases where it's discouraged to use `ValueFromPipelineByPropertyName` on other parameters, such as for a `New-*` cmdlet. In the cases where a cmdlet doesn't have a `-ResourceId` parameter, it's fine for them to add the `ValueFromPipelineByPropertyName` property to logical parameters to allow users to pipe in an object with the desired parameters as properties (either via a defined type or a hashtable).